src/compiler/checker.ts

```diff
@@ -7306,6 +7306,50 @@ namespace ts {
             }
         }
 
+        function isSuperCallExpression(n: Node): boolean {
+            return n.kind === SyntaxKind.CallExpression && (<CallExpression>n).expression.kind === SyntaxKind.SuperKeyword;
+        }
+
+        function findFirstSuperCall(n: Node): Node {
+            if (isSuperCallExpression(n)) {
+                return n;
+            }
+            else if (isFunctionLike(n)) {
+                return undefined;
+            }
+            return forEachChild(n, findFirstSuperCall);
+        }
+
+        /**
+         * Return a cached result if super-statement is already found.
+         * Otherwise, find a super statement in a given constructor function and cache the result in the node-links of the constructor
+         *
+         * @param constructor constructor-function to look for super statement
+         */
+        function getSuperCallInConstructor(constructor: ConstructorDeclaration): ExpressionStatement {

```

Comment:
The funny thing is that you don't actually use the returned value except to determine whether you've found a super call - so you should consider reducing your needs to `NodeLinks.hasSuperCall` or `NodeLinks.superCall`, but not both.